gdal/gdalinfo.xml

```diff
@@ -0,0 +1,224 @@
+            <option value="default" selected="true">Text</option>
+            <option value="json">Json</option>
+        </param>
+        <param name="subdataset" type="select" label="Create list of potential subdatasets" display="radio" multiple="False">
+            <option value="subd_y">Yes</option>
+            <option value="subd_n" selected="True">No</option>
+        </param>
+        <conditional name="settings">
+            <expand macro="gdal_advanced_params_select"/>
+            <when value="advanced">
+                <param name="mm" type="boolean" truevalue="-mm" falsevalue="" checked="false" help="-mm" label="Force computation of the actual min/max values for each band in the dataset."/>
+                <expand macro="gdal_param_stats"/>                    
+                <param name="approx_stats" type="boolean" truevalue="-approx_stats" falsevalue="" checked="false" help="-approx_stats" label="Read and display image statistics."/>
+                <param name="hist" type="boolean" truevalue="-hist" falsevalue="" checked="false" help="-hist" label="Report histogram information for all bands."/>
+                <param name="nogcp" type="boolean" truevalue="-nogcp" falsevalue="" checked="false" help="-nogcp" label="Suppress ground control points list printing."/>
+                <param name="nomd" type="boolean" truevalue="-nomd" falsevalue="" checked="false" help="-nomd" label="Suppress metadata printing."/>
+                <expand macro="gdal_param_norat"/>
+                <param name="noct" type="boolean" truevalue="-noct" falsevalue="" checked="false" help="-noct" label="Suppress printing of color table."/>
+                <param name="checksum" type="boolean" truevalue="-checksum" falsevalue="" checked="false" help="-checksum" label="Force computation of the checksum for each band in the dataset."/>
+                <param name="listmdd" type="boolean" truevalue="-listmdd" falsevalue="" checked="false" help="-listmdd" label="List all metadata domains available for the dataset."/>
+                <param name="proj4" type="boolean" truevalue="-proj4" falsevalue="" checked="false" help="-proj4" label="Report a PROJ.4 string."/> 
+                <param name="mdd" type="text" help="-mdd DOMAIN" label="Report metadata for the specified domain." value="all">
+                    <validator type="expression" message="mdd option can't be an empty value">value.strip()</validator>
+                </param>    
+            </when>
+        </conditional>
+    </inputs>
+
+    <outputs>
+        <data name="outputDefault" format="txt" label="Default info file">
+            <filter>format == "default"</filter>
+        </data>
+        <data name="outputJson" format="json" label="Json formated info file">
+            <filter>format == "json"</filter>
+        </data>
+        <data name="Subdatasets" format="tabular" from_work_dir="*_Subdatasets" label="Subdatasets from ${input1.name}">
```

Comment:
If these is a potential list of datasets, we need to create a collection here with `discover_datasets`

Reply:
There is just one file it's a misleading writing from me..